tests/test_tpt_reports.py

```diff
@@ -143,6 +144,40 @@ def test_domain_validation(mock_generate_reports):
             assert return_code == 2, "main() should return with error return code 2"
 
 
+@patch("tpt_reports.tpt_reports.report_gen")
+def test_generate_reports(mock_report_gen):
+    """Validate functionality for generate_reports()."""
+    mock_report_gen.return_value = True
+    return_val = tpt_reports.tpt_reports.generate_reports(
+        "test", "test", "cisa.gov", "./test_output", TEST_JSON_FILE
+    )
+    assert return_val is True, "generate_reports() failed to generate a report."
+
+
+@patch("tpt_reports.tpt_reports.report_gen")
+def test_generate_reports_bad_file_name(mock_report_gen):
+    """Validate functionality of generate_reports() when a bad filename is provided."""
+    mock_report_gen.return_value = False
+    return_val = tpt_reports.tpt_reports.generate_reports(
+        "test", "test", "cisa.gov", "./test_output", "nonexistent_file.json"
+    )
+    assert (
+        return_val is False
+    ), "generate_reports() returned a boolean True when a bad filename was provided."
```

Comment:
I'd reword the assertion error message slightly, since you are only validating that the return value is not `False` - there's no guarantee that this means it returned `True`:
```suggestion
    ), "generate_reports() failed to return a boolean False when a non-existent filename was provided."
```

Reply:
I committed the suggestion. Thank you.